### PR TITLE
Add conditional on can_write to prevent writing to the audit-db

### PR DESCRIPTION
**Pull request by adam-james-v**
Closes: #34915

Implement `can-write` method for the database model that returns false if the database in question is the Internal audit database.

This should prevent all users from modifying details about the audit db.

### CODE REVIEW

**Review by escherize — APPROVED** [review]
Looks good to me! It will clearly reject any can-write? check for the audit db in both arities.

**Comment by luizarakaki** [comment]
@adam-james-v tested renaming the db, triggering sync, scanning, discard values. Everything was blocked as expected.

But I could delete the Internal Metadata Database. I think we should block this too

File: enterprise/backend/test/metabase_enterprise/audit_app/api/database_test.clj [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+   [metabase.test :as mt]))
+
+(deftest audit-db-unmodifiable-test
+  (premium-features-test/with-premium-features #{:audit-app}
+    (audit-db-test/with-audit-db-restoration
+      (audit-db/ensure-audit-db-installed!)
+      (testing "Neither admin nor regular users can modify the audit database"
+        (doseq [[verb path] [[:post "database/%d/unpersist"]
+                             [:put "database/%d"]
+                             [:get "database/%d/syncable_schemas"]]
```
Comment (escherize) [inline enterprise/backend/test/metabase_enterprise/audit_app/api/database_test.clj:17]: I think you can add `[:delete "database/%d"]` to this list to get the failing repro

**Comment by escherize** [comment]
> But I could delete the Internal Metadata Database.

@luizarakaki Do you know which route you used? Was it `DELETE /api/database/:id`? That info helps make things easier to fix

**Comment by luizarakaki** [comment]
Pretty sure it is the DELETE method

**Review by luizarakaki — APPROVED** [review]
delete blocked as expected

**Review by escherize — APPROVED** [review]
✅

**Pull request merged**